utils.py

```diff
@@ -1,43 +1,59 @@
 #!/usr/bin/env python
 
 import sys
+import array
 import pygame
 
 import wx
 wx.App()
 
 import numpy as np
 
+from PIL import Image
+
 from skimage.color import rgb2gray
 from skimage.transform import resize
 from skimage.io import imread
+from skimage.util import img_as_float
 
 import matplotlib.pyplot as plt
 import matplotlib.image as mpimg
 
+SRC_W = 615 # 640
+SRC_H = 480
+SRC_D = 3
+OFFSET_X = 0 # 400
+OFFSET_Y = 0 # 240
 
 IMG_W = 200
 IMG_H = 66
+IMG_D = 3
 
 
 def take_screenshot():
     screen = wx.ScreenDC()
-    size = screen.GetSize()
-    bmp = wx.Bitmap(size[0], size[1])
+    bmp = wx.Bitmap(SRC_W, SRC_H)
     mem = wx.MemoryDC(bmp)
-    mem.Blit(0, 0, size[0], size[1], screen, 0, 0)
-    return bmp.GetSubBitmap(wx.Rect([0,0],[615,480]))
+    mem.Blit(0, 0, SRC_W, SRC_H, screen, OFFSET_X, OFFSET_Y)
+    return bmp
+
 
+arr = array.array('B', [0] * (SRC_W * SRC_H * SRC_D));
```

Comment:
Correct. That array allocation took a while, so moving it outside the method sped things up quite a bit. Any preferences for the class? Where to define it... what to call it.. etc? If not, I'll come up with something, but just want to avoid going round and round if we can hammer it out before I start.